### PR TITLE
Enhance JSONForms Preview

### LINKED ISSUE #15: Investigate long preview loading time
Preview takes a long time to load on the first call.

### LINKED ISSUE #16: Enhance preview workflow
- Check automatically if file is schema or uischema
- Enhance visual message, so the user knows, what to do next

### PR DESCRIPTION
**Pull request by TheZoker**
- Add automated check for schema or uiSchema, fixes #16
- Add loading screen to preview, partially fixes #15 

Note: Travis fails, because the file is to large (> 500 lines of code). This will be fixed in the next PR!

### CODE REVIEW

File: jsonforms-tooling-common/src/index.ts [hunk 1]
```diff
@@ -111,20 +113,32 @@ export const showPreview = async (editorInstance: any, firstSchemafileUri: any,
...
+      showMessage(err.message, 'err');
       return;
     }
   }
   // In both situations we still need the second schema file
+  // First we'll tell the user what to do next
+  try {
+    await editorInstance.window.showQuickPick(['Next'], editorInstance.QuickPickOptions = {
+      canSelectMany: false,
+      placeHolder: `It seems, that you selected the ${uiSchemaOrSchema}. Now  please select the ${otherSchema} file.`
```
Comment (edgarmueller) [inline jsonforms-tooling-common/src/index.ts]: Rephrase: You've selected a ${uiSchemaOrSchema} file. Continue by selecting a ${otherSchema} file.

File: jsonforms-tooling-common/src/index.ts [hunk 2]
```diff
@@ -232,14 +236,15 @@ const asyncGenerateUiSchema = async (editorInstance: any, path: string) => {
...
  * @param {function} callback forwards the current status to the caller
  */
-const validateJSONSchema = async (schema: Object) => {
+const validateUiSchema = async (schema: Object) => {
+  let valid = null;
   try {
     const ajv =  new Ajv();
-    ajv.compile(schema);
-    return true;
+    valid = ajv.validate(uiMetaSchema, schema);
```
Comment (edgarmueller) [inline jsonforms-tooling-common/src/index.ts]: Return here directly, no need for the `valid` local var.

Comment (TheZoker) [reply jsonforms-tooling-common/src/index.ts]: Done

**Comment by edgarmueller** [comment]
Thanks.

**Pull request merged**